blog/2021-07-17-introducing-metriql.md

```diff
@@ -0,0 +1,94 @@
+
+We have been also giving careful thoughts to this particular problem for a while. The way we think about metrics is not just about the definitions, instead, a better integration between data tools and data warehouses. Most companies already use a modern data warehouse and have adopted the data warehouse as the single source of truth. They ingest data in real-time, run ELT jobs directly in their data warehouse, run ad-hoc queries and get the results in low-latency, and even run ML jobs on them.
+
+If you’re using a data transformation tool such as dbt, you can model your data and expose it to your data tools but that’s not often enough for the data tools, we often found ourselves doing extra data modeling in data tools for metric definitions. This is primarily because most of the data tools introduce different ways to define metrics and you need to adopt it to be able to analyze the data reliably. Let's categorize the metric types under 3 category:
+
+### Aggregate measures
+In most cases, you usually have a numeric column in your database and define a metric with aggregation functions such as SUM, COUNT, or AVERAGE. If you use Tableau to analyze your sales data, you usually connect it to a table in your data warehouse that has a `Number of Sales` column, use Tableau’s [SUM](https://help.tableau.com/current/pro/desktop/en-us/calculations_aggregation.htm) function to define a default aggregation for that column and slice & dice the data. If your business stakeholders (usually the finance people in this case) don't know about the Tableau expressions, you need to create a workbook manually on behalf of them every time you change the data model. Even if you use the [SUM](https://www.w3schools.com/sql/sql_count_avg_sum.asp) function in ANSI SQL, Tableau doesn’t recognize it and you need to search and use the appropriate Tableau aggregate function to define the metric. 
+
+### Non-aggregate measures
+As your data model gets more advanced, you start to calculate the metrics such as LTV, and CAC. Customer acquisition cost usually defined as follows:
+
+```
+[Total Expenses Metric] / [Total Customers Metric]
+```
+
+If you use SQL to define this metric in your data model, you can’t just use an aggregation function such as SUM to calculate this metric. Instead, it should be calculated on the fly directly from your visualization tool from the source data. That’s why Tableau has [calculations](https://help.tableau.com/current/pro/desktop/en-us/calculations_calculatedfields_lod.htm), Metabase has [custom expressions](https://www.metabase.com/learn/questions/custom-expressions), PowerBI has [DAX](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-quickstart-learn-dax-basics). SQL is not enough anymore even though you can write the same expression in SQL, you need to learn these custom expression languages to be able to define your metrics, usually using their GUI.
```

Comment:
Instead, it should be calculated on the fly directly from your visualization tool from the source data  ( from from dogru mu?)

Reply:
directly from visualization tool pek dogru degil, in ad-hoc SQL query for the report that you're generating olabilir.

Reply:
```suggestion
If you use SQL to define this metric in your data model, you can’t just use an aggregation function such as SUM to calculate this metric. Instead, it should be calculated on the fly directly in an ad-hoc SQL query for the report you're generating. That’s why Tableau has [calculations](https://help.tableau.com/current/pro/desktop/en-us/calculations_calculatedfields_lod.htm), Metabase has [custom expressions](https://www.metabase.com/learn/questions/custom-expressions), PowerBI has [DAX](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-quickstart-learn-dax-basics). SQL is not enough anymore even though you can write the same expression in SQL, you need to learn these custom expression languages to be able to define your metrics, usually using their GUI.
```